streamer/src/streamer.rs

```diff
@@ -184,6 +189,92 @@ pub fn receiver(
         .unwrap()
 }
 
+fn recv_loop2(
+    socket: &UdpSocket,
+    exit: Arc<AtomicBool>,
+    packet_batch_sender: &TxPacketBatchSender,
+    recycler: &PacketBatchRecycler,
+    stats: &StreamerReceiveStats,
+    coalesce_ms: u64,
+    use_pinned_memory: bool,
+    in_vote_only_mode: Option<Arc<AtomicBool>>,
+) -> Result<()> {
+    loop {
+        let mut packet_batch = if use_pinned_memory {
+            PacketBatch::new_with_recycler(recycler.clone(), PACKETS_PER_BATCH, stats.name)
+        } else {
+            PacketBatch::with_capacity(PACKETS_PER_BATCH)
+        };
+        loop {
+            // Check for exit signal, even if socket is busy
+            // (for instance the leader transaction socket)
+            if exit.load(Ordering::Relaxed) {
+                return Ok(());
+            }
+
+            if let Some(ref in_vote_only_mode) = in_vote_only_mode {
+                if in_vote_only_mode.load(Ordering::Relaxed) {
+                    sleep(Duration::from_millis(1));
+                    continue;
+                }
+            }
+
+            if let Ok(len) = packet::recv_from(&mut packet_batch, socket, coalesce_ms) {
+                if len > 0 {
+                    let StreamerReceiveStats {
+                        packets_count,
+                        packet_batches_count,
+                        full_packet_batches_count,
+                        max_channel_len,
+                        ..
+                    } = stats;
+
+                    packets_count.fetch_add(len, Ordering::Relaxed);
+                    packet_batches_count.fetch_add(1, Ordering::Relaxed);
+                    max_channel_len.fetch_max(packet_batch_sender.len(), Ordering::Relaxed);
+                    if len == PACKETS_PER_BATCH {
+                        full_packet_batches_count.fetch_add(1, Ordering::Relaxed);
+                    }
+
+                    let packet_batch = TxPacketBatch::from_packet_batch(packet_batch);
```

Comment:
This while `recv_loop2() & receiver2()` is a kind of a hack I made at the moment in lieu of deferring a design decision to review.

The question is whether we want to use the new data structure `TxPacketBatch` all throughout the codebase (Tx's, shreds, gossip), or keep its' use limited to the Tx path. As I have it coded now, I keep the change limited to the Tx processing path, and this is a shim for minimizing code duplication (can probably get something cleaner in the end result).

So, the direction question that I'd like feedback on is: Should we keep the existing PacketBatch data structure around for uses where we will continue to use UDP (and thus keep the 1232 byte restriction), or do we want to go all in and use the new data structure everywhere?